src/Microsoft.ML.AutoML/AutoMLExperiment/IPerformanceMonitor.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading.Tasks;
+using System.Timers;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.AutoML
+{
+    public interface IPerformanceMonitor : IDisposable
```

Comment:
>IPerformanceMonitor [](http://example.com/codeflow?start=21&length=19)

Do we expect extending this interface later? I am asking to know if it is better to make it abstract class if we run on .NET Framework and avoid breaking future change.